core/handlers/v2/logs_handler.go

```diff
@@ -35,15 +38,27 @@ func (this *LogsHandler) Get(w http.ResponseWriter, req *http.Request, next http
 
 	queryParams := req.URL.Query()
 	limitQuery, err := strconv.Atoi(queryParams.Get("limit"))
-	if err == nil {
-		logs = this.Hoverfly.GetFilteredLogsView(limitQuery)
+
```

Comment:
I think the following (pseudocode) might be less verbose:
```
// query is an object will a set of filters, or empty if query param is not present
query = parse(queryParams) 
logs = this.Hoverfly.getLogs(query) 
logsView = logsViewMapper.toView(logs)
```
Obviously, I don't know how you implement this in Go, but this would be how I do it in Java.